core/src/client/async_client/mod.rs

```diff
@@ -232,6 +248,16 @@ impl Client {
 		*err_lock = next_state;
 		err
 	}
+
+	/// Completes when the client is disconnected or the client's background task encountered an error.
+	///
+	/// # Cancel safety
+	///
+	/// This method is cancel safe. Once the client is disconnected, it stays disconnected forever and all
+	/// future calls to this method will return immediately.
```

Comment:
I love the thought to cancel safety, though most of the text isn't really related to cancel safety so I'd move it up; something like:

```suggestion
	/// Completes when the client is disconnected or the client's background task encountered an error. If the client is already disconnected, the future produced by this method will complete immediately.
	///
	/// # Cancel safety
	///
	/// This method is cancel safe.
```